petastorm/codecs.py

```diff
@@ -20,11 +20,15 @@
 NOTE: Due to the way unischema is stored alongside dataset (with pickling), changing any of these codecs class names
 and fields can result in reader breakages.
 """
-
+import sys
 from abc import abstractmethod
 from io import BytesIO
 
-import cv2
+try:
+    import cv2
+except ImportError:
+    pass
```

Comment:
I think rather than using the assert below on `sys.modules` you can have a global variable `OPENCV_AVAILABLE` here to check whether the ImageCodec can be used